esx_service/utils/auth_data.py

```
@@ -561,10 +568,46 @@ def handle_upgrade_1_1_to_1_2(self):
+                id = r['id']
+                ds_url = r['default_datastore_url']
+                privilege = (id, ds_url, 1, 0, 0)
+                self.conn.execute("INSERT OR IGNORE INTO privileges(tenant_id, datastore_url, allow_create, max_volume_size, usage_quota) VALUES (?, ?, ?, ?, ?)",
+                                  privilege)
+            logging.debug("handle_upgrade_1_1_to_1_2: Insert privilege to default_datastore in privileges table")
+
+            cur = self.conn.execute("SELECT * FROM tenants WHERE id = ?",
+                                    (auth_data_const.DEFAULT_TENANT_UUID,)
+                                   )
+
+            result = cur.fetchall()
+            logging.debug("handle_upgrade_1_1_to_1_2: Check DEFAULT tenant exist")
+            if result:
+                # _DEFAULT tenant exists
+                # insert full access privilege to "__ALL_DS" for "_DEFAULT" tenant
+                all_ds_privilege = (auth_data_const.DEFAULT_TENANT_UUID, auth_data_const.ALL_DS_URL, 1, 0, 0)
+                self.conn.execute("INSERT INTO privileges(tenant_id, datastore_url, allow_create, max_volume_size, usage_quota) VALUES (?, ?, ?, ?, ?)",
+                                  all_ds_privilege)
+                logging.debug("handle_upgrade_1_1_to_1_2: Insert privilege to __ALL_DS for _DEFAULT tenant in privileges table")
+                # remove access privilege to "DEFAULT_DS"
+                self.conn.execute("DELETE FROM privileges WHERE tenant_id = ? AND datastore_url = ?",
+                                [auth_data_const.DEFAULT_TENANT_UUID, auth_data_const.DEFAULT_DS_URL])
+                logging.debug("handle_upgrade_1_1_to_1_2: Remove privilege to _DEFAULT_DS for _DEFAULT tenant in privileges table")
             self.conn.commit()
             return None
         except sqlite3.Error as e:
-            error_msg = "Error when upgrading auth DB VMs table"
+            error_msg = "Error when upgrading auth DB table"
```

please do not hide the exception but add it to the message(not only logging,error)